tests/unit-core/test-api-value-type.c

```diff
@@ -124,6 +124,33 @@ main (void)
     jerry_release_value (object_bigint);
   }
 
+  if (jerry_is_feature_enabled (JERRY_FEATURE_REALM))
+  {
+    jerry_value_t new_realm = jerry_create_realm ();
+    jerry_value_t old_realm = jerry_set_realm (new_realm);
+
+    jerry_type_t new_realm_type = jerry_value_get_type (new_realm);
+    TEST_ASSERT (new_realm_type == JERRY_TYPE_OBJECT);
+
+    jerry_value_t new_realm_this = jerry_realm_get_this (new_realm);
+    jerry_type_t new_realm_this_type = jerry_value_get_type (new_realm_this);
```

Comment:
By default the `this` is the same as `realm`.

Reply:
Okay and? I've explicitly testing it's type.